gcloud/translate/client.py

```diff
@@ -0,0 +1,199 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Client for interacting with the Google Cloud Translate API."""
+
+
+import httplib2
+import six
+
+from gcloud._helpers import _to_bytes
+from gcloud.translate.connection import Connection
+
+
+ENGLISH_ISO_639 = 'en'
+"""ISO 639-1 language code for English."""
+
+
+def _zip_assert(values1, values2):
+    """Zips two iterables and asserts they are the same length.
+
+    :type values1: list
+    :param values1: Iterable (with a length) to be zipped.
+
+    :type values2: list
+    :param values2: Iterable (with a length) to be zipped.
+
+    :rtype: iterator
+    :returns: Zip iterator for each pair of values.
+    :raises: :class:`ValueError <exceptions.ValueError>` if the number of
+             values in each list is not the same.
+    """
+    if len(values1) != len(values2):
+        raise ValueError('Expected iterations to have same length',
+                         values1, values2)
+
+    return six.moves.zip(values1, values2)
+
+
+class Client(object):
+    """Client to bundle configuration needed for API requests.
+
+    :type key: str
+    :param key: The key used to send with requests as a query
+                parameter.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: (Optional) HTTP object to make requests. If not
+                 passed, an :class:`httplib.Http` object is created.
+    """
+
+    def __init__(self, key, http=None):
+        self.key = key
+        if http is None:
+            http = httplib2.Http()
+        self.connection = Connection(http=http)
+
+    def get_languages(self, target_language=None):
+        """Get list of supported languages for translation.
+
+        Response
+
+        See: https://cloud.google.com/translate/v2/\
+        discovering-supported-languages-with-rest
+
+        :type target_language: str
+        :param target_language: (Optional) The language used to localize
+                                returned language names.
+
+        :rtype: list
+        :returns: List of dictionaries. Each dictionary contains a supported
+                  ISO 639-1 language code (using the dictionary key
+                  ``language``). If ``target_language`` is passed, each
+                  dictionary will also contain the name of each supported
+                  language (localized to the target language).
+        """
+        query_params = {'key': self.key}
+        if target_language is not None:
+            query_params['target'] = target_language
+        response = self.connection.api_request(
+            method='GET', path='/languages', query_params=query_params)
+        return response.get('data', {}).get('languages', ())
+
+    def detect_language(self, *values):
```

Comment:
In Datastore, we had `.get` which accepted a key or sequence of keys. Then NDB swapped this all to `.get_multi` (with no more `.get`).

Is there a concern about accepting both a string or a sequence of strings? I can't find anything in style guides talking about this. Some on SO mention the `*args` approach as well as the "check if it's a the type you expected a list of" (ie, string when you expected a list of strings) and return from there.

IMO, I think the best bet here is that pattern:

```python
def detect_language(self, strings):
  if isinstance(strings, six.string_types):
    return self.detect_language([strings])[0]

  # Continue as though it were multiple strings...
  return api_request(...)
```

http://stackoverflow.com/questions/12603597/writing-methods-that-accept-either-a-single-item-or-an-iterable-in-python